### PR TITLE
docs: Add a troubleshooting for ipa installation

### PR DESCRIPTION
**Pull request by KazuCocoa**
## Proposed changes

A library we use in unzip generates wrong file names if they were unicode in `ipa`.
It was because the zip (ipa) had no proper flag.
https://github.com/appium/appium/issues/14100#issuecomment-653733643

~~I also tried to unzip with https://www.npmjs.com/package/unzipper.~~
~~It could unzip properly but I'm not sure the package could be a proper replacement.~~ <= the unzipped did not keep permissions, so it could not be an alternative for now.
So, for now, let us address the file name stuff in troubleshooting

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
@KazuCocoa If you say that unzipper can guess file names encoding properly then it probably makes sense to try it for appium@beta. .ipa files are packed automatically by Xcode, so there is no easy solution for users unless Apple does the fix to its zip libs :(

**Comment by mykola-mokhnach** [comment]
@jlipps What do you think?

**Comment by KazuCocoa** [comment]
Updated the description. I addressed unzipper did not keep file permissions.

File: docs/en/writing-running-appium/other/troubleshooting.md [hunk 1]
```diff
@@ -141,6 +141,14 @@ own items in the SDK manager. Make sure you install the build-tools and platform
     - If it returns `69`, you should call `xcodebuild -runFirstLaunch` once to prevent the root error, `Install Started
 Install Failed: Error Domain=DVTDownloadableErrors Code=3 "The authorization was denied since no user interaction was possible." UserInfo={AuthorizationErrorCode=-60007, NSLocalizedDescription=The authorization was denied since no user interaction was possible.}`
     - example: https://github.com/appium/WebDriverAgent/pull/165#discussion_r281458167
+* Make sure your `.ipa` had no Unicode file names
+    - Appium unzip `.ipa` to install packed the application under test, but [the macOS's Archive Utils has an issue](https://stackoverflow.com/questions/13261347/correctly-decoding-zip-entry-file-names-cp437-utf-8-or). So, unzip generates could generate wrong file names.
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/other/troubleshooting.md:145]: `So, unzip generates could generate wrong file names -> As a result unzipped file names could contain invalid characters`

Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/other/troubleshooting.md:145]: Appium unzips .ipa packages to be able to install the app under test, but there a known issue with Mac OS ZIP library

File: docs/en/writing-running-appium/other/troubleshooting.md [hunk 2]
```diff
@@ -141,6 +141,14 @@ own items in the SDK manager. Make sure you install the build-tools and platform
...
+    - As another workaround, you can follow below step.
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/other/troubleshooting.md:146]: As a workaround you could follow the below steps

File: docs/en/writing-running-appium/other/troubleshooting.md [hunk 3]
```diff
@@ -141,6 +141,14 @@ own items in the SDK manager. Make sure you install the build-tools and platform
...
+      1. Rename `your_test.ipa` to `your_test.zip`
+      2. Unzip it (by Archive Utils by Apple for example). Then, `Payload/your_test.app` is generated
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/other/troubleshooting.md:148]: using ..., for example

Comment (mykola-mokhnach) [reply docs/en/writing-running-appium/other/troubleshooting.md:148]: Make sure that `Payload/your_test.app` is present in the extracted folder

File: docs/en/writing-running-appium/other/troubleshooting.md [hunk 4]
```diff
@@ -141,6 +141,14 @@ own items in the SDK manager. Make sure you install the build-tools and platform
...
+      3. Set `your_test.app` as `app` capability
+        - Or zip the `your_test.app` again properly
```
Comment (mykola-mokhnach) [inline docs/en/writing-running-appium/other/troubleshooting.md:150]: here we need to provide a proper command line to do it

**Comment by KazuCocoa** [comment]
Close since https://github.com/appium/appium-support/pull/193 's force utf8 encode can solve this issue

**Pull request closed without merging**